### PR TITLE
I668 doc cover rst

### PR DESCRIPTION
**Pull request by fcooper8472**
Addresses #668

### CODE REVIEW

File: pints/noise.py [hunk 1]
```diff
@@ -7,7 +7,6 @@
 #  For licensing information, see the LICENSE file distributed with the PINTS
 #  software package.
 #
-import numpy as np
```
[the remark this answers is not in the document]
Comment (ben18785) [reply pints/noise.py]: Fergz explained to me in person -- I'm happy now

File: docs/source/toy/neals_funnel.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+*****************************
+Twisted Gaussian distribution
```
[the remark this answers is not in the document]
Comment (ben18785) [reply docs/source/toy/neals_funnel.rst]: I'll change this...

**Comment by fcooper8472** [comment]
Thanks @ben18785 

There's a fair bit of technical stuff so I'll wait for @MichaelClerx comments on that

File: docs/source/toy/neals_funnel.rst [hunk 3]
```diff
@@ -0,0 +1,7 @@
+*****************************
+NEALS FUNNEL DISTRIBUTION
```
Comment (MichaelClerx) [inline docs/source/toy/neals_funnel.rst]: Don't need all-caps here!
Also *s should have same length as the words below (or sphinx breaks)

File: run-tests.py [hunk 4]
```diff
@@ -148,6 +152,163 @@ def doctest_examples_readme():
...
+    check_exposed_symbols(pints.io, [], doc_symbols)
+    check_exposed_symbols(pints.noise, [], doc_symbols)
+    check_exposed_symbols(pints.plot, [], doc_symbols)
+    check_exposed_symbols(pints.toy, [], doc_symbols)
+
+    print('All classes and methods are documented in an RST file, and all '
+          'public interfaces are clean.')
+
+
+def check_exposed_symbols(_module, _submodule_names, _doc_symbols):
```
Comment (MichaelClerx) [inline run-tests.py]: Function arguments should never have initial underscores

File: run-tests.py [hunk 5]
```diff
@@ -148,6 +152,163 @@ def doctest_examples_readme():
...
+    """
+    Check _module for any classes and methods not contained in _doc_symbols,
```
Comment (MichaelClerx) [inline run-tests.py]:
```
not contained in ``doc_symbols``
```
(same for remaining)

File: run-tests.py [hunk 6]
```diff
@@ -148,6 +152,163 @@ def doctest_examples_readme():
...
+    and check for any modules not contained in _submodule_names.
+
+    :param _module: the module to check
```
Comment (MichaelClerx) [inline run-tests.py]: This is cool, but we're not using it at the moment and not sure if our template supports it. If we start using this it'll have to be a pints-wide change in a separate commit. For now please do:

```
Arguments:

``module``
    The module to check
``submodule_names``
    List of submodules exposed by ``module``
```
etc.

(that creates a definition list)

File: run-tests.py [hunk 7]
```diff
@@ -148,6 +152,163 @@ def doctest_examples_readme():
...
+    :param _submodule_names: list of submodules exposed by _module
+    :param _doc_symbols: dictionary containing lists of documented classes and
+     functions
+    """
+
+    import inspect
+    exposed_symbols = [x for x in dir(_module) if not x.startswith('_')]
+    symbols = [getattr(_module, exposed_symbols[x]) for x in
```
Comment (MichaelClerx) [inline run-tests.py]: why not just `[getattr(module, x) for x in exposed_symbols]` ?

File: run-tests.py [hunk 8]
```diff
@@ -148,6 +152,163 @@ def doctest_examples_readme():
...
+               range(len(exposed_symbols))]
+
+    classes = [x for x in symbols if inspect.isclass(x)]
```
Comment (MichaelClerx) [inline run-tests.py]: Niiiiiiice

File: run-tests.py [hunk 9]
```diff
@@ -148,6 +152,163 @@ def doctest_examples_readme():
...
+        print('FAILED')
+        sys.exit(1)
+
+
+def get_all_documented_symbols():
+    """
+    Recursively traverse docs/source and identify all autoclass and
+    autofunction declarations.
+
+    :return: A dict containing a list of classes and a list of functions
```
Comment (MichaelClerx) [inline run-tests.py]: Same as for `:param:` above!
At the moment we just write "Returns: A dict containing..."

Comment (MichaelClerx) [reply run-tests.py]: Or "Returns a dict..."

File: run-tests.py [hunk 10]
```diff
@@ -148,6 +152,163 @@ def doctest_examples_readme():
...
+    """
+
+    doc_files = []
+    for root, dirs, files in os.walk(os.path.join('docs', 'source')):
+        for file in files:
+            if file.endswith('.rst'):
+                doc_files.append(os.path.join(root, file))
+
+    # Regular expressions to find autoclass and autofunction specifiers
+    regex_module = re.compile(r'\.\.\s*\S*module\:\:\s*(\S+)')
```
Comment (MichaelClerx) [inline run-tests.py]: Maybe move comment down 2 lines, and add a separate comment explaining about the module regex, and how it also captures currentmodule directives

File: run-tests.py (hunk 4 above)
Comment (fcooper8472) [reply run-tests.py]: Why not?

File: run-tests.py (hunk 7 above)
Comment (fcooper8472) [reply run-tests.py]: D'oh!

**Comment by fcooper8472** [comment]
@MichaelClerx pretty sure I've hit all your comments now.

File: run-tests.py (hunk 4 above)
Comment (MichaelClerx) [reply run-tests.py]: They're outward facing. Not hidden. Underscores are for stuff you want to hide

**Pull request merged**